allow to change prefix, and define default values for properties

HellButcher opened the pull request.

cstettler commented:
Hi @HellButcher, thank you for your pull request and sorry for the delay in answering. May I kindly ask you to quickly describe the use cases you want to support with these changes? Just to be sure to understand them correctly. Thank you!

HellButcher commented:
> Hi @HellButcher, thank you for your pull request and sorry for the delay in answering. May I kindly ask you to quickly describe the use cases you want to support with these changes? Just to be sure to understand them correctly. Thank you!

I have edited the description

cfranzen commented:
@cstettler any chance that we can get this PR merged?

cstettler commented:
Hi @HellButcher and @cfranzen, I have merged the PR - sorry for the delay. While revisiting the result, I am not so sure about the feature of supporting additional attributes. Before releasing a new version, I would like to clarify the use case and also the implementation. I was initially trying to support additional attributes by default, but failed to do so (see the two ignored tests). Could you please re-iterate on the specific use case(s) you want to support, and also explain your take on the implementation? Thank you!

HellButcher commented:
Hi @cstettler.
I have totally forgotten the "additional-attributes" usecase. 
I added it to the description of the PR). and updated the tests:
- https://github.com/cstettler/thymeleaf-component-dialect/pull/2

My take on the implementation was to make it EXPCICIT on which element in the template to place the additional attributes:
For example a component that wraps an `<input>` in an other element, may want to explicitly state that the `<input>` should receive the additional attributes.

Also: a component template may have several layers of `<th:block>` (there is at least a `<th:block>` at the root level). In an hypothetical implementation, that may implicitlz automatically place the additional attributes at the root-element of the template, the additional attributes would get lost (at least not rendered):

The implementation regarding additional-attributes is as follows:
- get the additional attributes (as before)
- get the template fragment sub-tree (I moved this out of the `fillSlots` method)
- add the additional-attributes to each element having the `pl:pass-additional-attributes` attribute
- OR if there was no `pl:pass-additional-attributes` attribute: add an opening `<th:block>` with the additional-attributes (as before)
- call `fillSlots` (as before) and add it to the result (as before)
- close the `</th:block>` if it was opened before (as before)
